src/Engine/Manager/SoundManager.cpp

```diff
@@ -70,10 +72,15 @@ void SoundManager::CheckError(PaError err) {
 
 void SoundManager::Update(float deltaTime) {
 
+    std::vector<Component::Listener*> listeners = GetListeners();
```

Comment:
Consider using const ref to avoid copying the vector (if `GetListeners` returns it, that is).